### ISSUE TITLE
WhitespaceAround should have an option allow double-brace initialization

### ISSUE THREAD

**Issue by kilotaras** [checkstyle/checkstyle#2838]
See following snippet

```
$ javac Fail.java

$ cat Fail.java
import java.util.Properties;

public class Fail {
    public Fail() {
        new Properties() {{
            setProperty("double curly braces", "are not a style error");
        }};
    }
}

$ cat fail.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="WhitespaceAround" />
    </module>
</module>

$ java -jar checkstyle-6.14.1-all.jar -c ./fail.xml Fail.java
Starting audit...
[ERROR] /home/tboiko/dev/tmp/checkstyle/Fail.java:5:27: '{' is not followed by whitespace. [WhitespaceAround]
[ERROR] /home/tboiko/dev/tmp/checkstyle/Fail.java:5:27: '{' is not preceded with whitespace. [WhitespaceAround]
[ERROR] /home/tboiko/dev/tmp/checkstyle/Fail.java:7:10: '}' is not followed by whitespace. [WhitespaceAround]
[ERROR] /home/tboiko/dev/tmp/checkstyle/Fail.java:7:10: '}' is not preceded with whitespace. [WhitespaceAround]
Audit done.
Checkstyle ends with 4 errors
```

As this is a common pattern it would be nice to not raise an error by checkstyle.

Expected - either no error or at least an option to ignore those kind of errors.

As a side note - it would be nice to have that link in README.

**Comment by romani** [comment]
@kilotaras , please report issue with all details http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug

**Comment by kilotaras** [comment]
```
$ javac Fail.java                                                                                                                                                                                                             10:49:58

$ cat Fail.java                                                                                                                                                                                                               10:49:58
import java.util.Properties;

public class Fail {
    public Fail() {
        new Properties() {{
            setProperty("double curly braces", "are not a style error");
        }};
    }
}

$ cat fail.xml                                                                                                                                                                                                                10:49:58
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="WhitespaceAround" />
    </module>
</module>

$ java -jar checkstyle-6.14.1-all.jar -c ./fail.xml Fail.java                                                                                                                                                                 10:49:58
Starting audit...
[ERROR] /home/tboiko/dev/tmp/checkstyle/Fail.java:5:27: '{' is not followed by whitespace. [WhitespaceAround]
[ERROR] /home/tboiko/dev/tmp/checkstyle/Fail.java:5:27: '{' is not preceded with whitespace. [WhitespaceAround]
[ERROR] /home/tboiko/dev/tmp/checkstyle/Fail.java:7:10: '}' is not followed by whitespace. [WhitespaceAround]
[ERROR] /home/tboiko/dev/tmp/checkstyle/Fail.java:7:10: '}' is not preceded with whitespace. [WhitespaceAround]
Audit done.
Checkstyle ends with 4 errors.
```

Expected - either no error or at least an option to ignore those kind of errors.

As a side note - it would be nice to have that link in README.

**Comment by romani** [comment]
thanks a lot for good report.

**Comment by romani** [comment]
@kilotaras , if you interested to resolve this issue quickly , you are welcome with contribution, all logic is https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java , we have UTs with 100%  coverage. A lot of instructions on how to start development in checkstyle http://checkstyle.sourceforge.net/beginning_development.html

**Comment by romani** [comment]
This Check is used in Google style guide - http://checkstyle.sourceforge.net/google_style.html#a4.6.2.
This issue is going to grey area of Google style guide - https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace

> a single ASCII space also appears in the following places only.

ONLY!!!

> Before any open curly brace ({), with two exceptions:
    @SomeAnnotation({a, b}) (no space is used)
    String[][] x = {{"foo"}}; (no space is required between {{, by item 8 below)

our case is not specified, so it should not be allowed.
But "{{" (double-brace) initialization ability become well known and popular ability only recently. So author of Google Style guide might not know it or was not aware of it.

Example in guava - https://github.com/google/guava/blob/master/guava-tests/test/com/google/common/collect/MapsTest.java#L208

Unfortunately Google does not want to discuss any updates in his style. So we need to make decision ourself.

Resolution: we will skip this case by default in Check (no options is required or now). 
If any complains appear from users we will create an option. Only by user request with explanation. Extra options in Check is responsibility to keep it to keep compatibility, so we try to avoid creation of new options as much as possible/reasonable.

**Comment by romani** [comment]
fix is merged.

**Comment by JoshMcCullough** [comment]
I'm still seeing ...

>[ant:checkstyle] [ERROR] /...:19:13: 'block' child has incorrect indentation level 12, expected level should be 16. [Indentation]
>[ant:checkstyle] [ERROR] /...:20:9: 'block rcurly' has incorrect indentation level 8, expected level should be 12. [Indentation]

With this code:

```
public class SomeTest
{
    @Test
    public void test()
    {
        final var x = new Thing()
        {{
            setSomething(true); // <-- 'block' child has incorrect indentation level 12, expected level should be 16. [Indentation]
        }};  // <-- 'block rcurly' has incorrect indentation level 8, expected level should be 12. [Indentation]
    }
}
```

I looked at the [docs](https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheck.html) but didn't see how to allow this. What am I missing? Thanks.

**Comment by strkkk** [comment]
@JoshMcCullough this issue is not about indentation. Please submit new issue in proper format if something looks like a bug

**Comment by JoshMcCullough** [comment]
@strkkk Gotcha, submitted #11307.
